sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -44,55 +44,36 @@ internal class PartitionedDownloader
             _client = client;
 
             // Set _maxWorkerCount
-            if (transferOptions.MaximumConcurrency.HasValue)
+            if (transferOptions.MaximumConcurrency.HasValue
+                && transferOptions.MaximumConcurrency > 0)
```

Comment:
Nit - you can skip the `HasValue` checks with these comparisons because [C# makes them false when null](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types#lifted-operators).

Reply:
fixed.

Reply:
Actually, never mind, this is causing a lot of unit tests to fail, and I don't have time to look into it.